subsys/bluetooth/controller/ll_sw/nordic/lll/lll_adv_pdu.h

```diff
@@ -116,6 +120,9 @@ static inline void lll_adv_sync_data_release(struct lll_adv_sync *lll)
 static inline void lll_adv_sync_data_enqueue(struct lll_adv_sync *lll,
 					     uint8_t idx)
 {
+#if defined(CONFIG_BT_CTLR_ADV_SYNC_PBACK)
+	lll_adv_sync_update_pback(lll, idx);
```

Comment:
It's convenient to update AuxPtr here because it guarantees that it will be updated every time new data is enqueued and in ULL context we do not need to care about timing (too much). It also does not make any difference for the caller because it still enqueues data as usual and has no side-effects. I can add a comment explaining why this is done here. So I can add short comment to explain why this is done here, but keep function name to be consistent with other calls.

Reply:
In my humble opinion, convenience is not enough to make a function to have hidden side effects. It should be "visible" on the caller site what function actually does. Reader should not be forced to check each function call to see if there is something unexpected done inside.